tests/data_context/store/test_datasource_store.py

```diff
@@ -0,0 +1,158 @@
+        execution_engine={
+            "class_name": "PandasExecutionEngine",
+            "module_name": "great_expectations.execution_engine",
+        },
+        data_connectors={
+            "tripdata_monthly_configured": {
+                "class_name": "ConfiguredAssetFilesystemDataConnector",
+                "module_name": "great_expectations.datasource.data_connector",
+                "base_directory": "/path/to/trip_data",
+                "assets": {
+                    "yellow": {
+                        "class_name": "Asset",
+                        "module_name": "great_expectations.datasource.data_connector.asset",
+                        "pattern": r"yellow_tripdata_(\d{4})-(\d{2})\.csv$",
+                        "group_names": ["year", "month"],
+                    }
+                },
+            }
+        },
+    )
+
+
+def test_datasource_store_with_bad_key_raises_error(
+    empty_datasource_store: DatasourceStore, datasource_config: DatasourceConfig
+) -> None:
+    store: DatasourceStore = empty_datasource_store
+
+    error_msg: str = "key must be an instance of StringKey"
+
+    with pytest.raises(TypeError) as e:
+        store.set(key="my_bad_key", value=datasource_config)
+    assert error_msg in str(e.value)
+
+    with pytest.raises(TypeError) as e:
+        store.get(key="my_bad_key")
+    assert error_msg in str(e.value)
+
+
+def test_datasource_store_retrieval(
+    empty_datasource_store: DatasourceStore, datasource_config: DatasourceConfig
+) -> None:
+    store: DatasourceStore = empty_datasource_store
+
+    key: StringKey = StringKey(key="my_datasource")
+    store.set(key=key, value=datasource_config)
+    res: DatasourceConfig = store.get(key=key)
+
+    assert isinstance(res, DatasourceConfig)
+    assert DatasourceConfigSchema().dump(res) == datasource_config.to_dict()
+
+
+def test_datasource_store_retrieval_cloud_mode(
+    datasource_config: DatasourceConfig,
+    ge_cloud_base_url: str,
+    ge_cloud_access_token: str,
+    ge_cloud_organization_id: str,
+) -> None:
+    ge_cloud_store_backend_config: dict = {
+        "class_name": "GeCloudStoreBackend",
+        "ge_cloud_base_url": ge_cloud_base_url,
+        "ge_cloud_resource_type": "datasource",
+        "ge_cloud_credentials": {
+            "access_token": ge_cloud_access_token,
+            "organization_id": ge_cloud_organization_id,
+        },
+        "suppress_store_backend_id": True,
+    }
+
+    store: DatasourceStore = DatasourceStore(
+        store_name="my_cloud_datasource_store",
+        store_backend=ge_cloud_store_backend_config,
+    )
+
+    key: GeCloudIdentifier = GeCloudIdentifier(
+        resource_type="datasource", ge_cloud_id="foobarbaz"
+    )
+
+    with patch("requests.patch", autospec=True) as mock_patch:
+        type(mock_patch.return_value).status_code = PropertyMock(return_value=200)
+
+        store.set(key=key, value=datasource_config)
+
+        mock_patch.assert_called_with(
```

Comment:
generally love this, can we leverage the datasource_config() fixture to avoid duplicating the payload here?

Reply:
that's a great call! done!